flink-connector-mongodb/src/test/java/org/apache/flink/connector/mongodb/table/MongoDynamicTableSinkITCase.java

```diff
@@ -0,0 +1,510 @@
+
+        List<Document> actual = coll.find().into(new ArrayList<>());
+
+        assertThat(actual).isEqualTo(expected);
+    }
+
+    @Test
+    public void testSinkWithReservedId() throws Exception {
+        String database = "test";
+        String collection = "sink_with_reserved_id";
+
+        TableEnvironment tEnv = TableEnvironment.create(EnvironmentSettings.inStreamingMode());
+
+        tEnv.executeSql(
+                String.format(
+                        "CREATE TABLE mongo_sink ("
+                                + "_id STRING NOT NULL,\n"
+                                + "f1 STRING NOT NULL,\n"
+                                + "PRIMARY KEY (_id) NOT ENFORCED\n"
+                                + ")\n"
+                                + "WITH (%s)",
+                        getConnectorSql(database, collection)));
+
+        ObjectId objectId = new ObjectId();
+        tEnv.fromValues(row(objectId.toHexString(), "r1"), row("str", "r2"))
+                .executeInsert("mongo_sink")
+                .await();
+
+        MongoCollection<Document> coll =
+                mongoClient.getDatabase(database).getCollection(collection);
+
+        List<Document> actual = new ArrayList<>();
+        coll.find(Filters.in("_id", objectId, "str")).into(actual);
+
+        Document[] expected =
+                new Document[] {
+                    new Document("_id", objectId).append("f1", "r1"),
+                    new Document("_id", "str").append("f1", "r2")
+                };
+        assertThat(actual).containsExactlyInAnyOrder(expected);
+    }
+
+    @Test
+    public void testSinkWithoutPrimaryKey() throws Exception {
+        String database = "test";
+        String collection = "sink_without_primary_key";
+
+        TableEnvironment tEnv = TableEnvironment.create(EnvironmentSettings.inStreamingMode());
+
+        tEnv.executeSql(
+                String.format(
+                        "CREATE TABLE mongo_sink (" + "f1 STRING NOT NULL\n" + ")\n" + "WITH (%s)",
+                        getConnectorSql(database, collection)));
+
+        tEnv.fromValues(row("d1"), row("d1")).executeInsert("mongo_sink").await();
+
+        MongoCollection<Document> coll =
+                mongoClient.getDatabase(database).getCollection(collection);
+
+        List<Document> actual = new ArrayList<>();
+        coll.find().into(actual);
+
+        assertThat(actual).hasSize(2);
+        for (Document doc : actual) {
+            assertThat(doc.get("f1")).isEqualTo("d1");
+        }
+    }
+
+    @Test
+    public void testSinkWithNonCompositePrimaryKey() throws Exception {
+        String database = "test";
+        String collection = "sink_with_non_composite_pk";
+
+        Instant now = Instant.now();
+        List<Expression> testValues =
+                Collections.singletonList(
+                        row(1L, true, "ABCDE", 12.12d, 4, Timestamp.from(now), now));
```

Comment:
```suggestion
                        row(2L, true, "ABCDE", 12.12d, 4, Timestamp.from(now), now));
```
Let's use some value that doesnt happen to coincide with the generated(?) `_id`.